execution-plane/arcon/src/lib.rs

```diff
@@ -8,6 +8,7 @@
 //! [arcon_codegen] to generate the Rust code.
 
 #![feature(unboxed_closures)]
+#![feature(fn_traits)]
```

Comment:
I'm not up to date with these nightly features. So my questions: (1) Are they necessary? (2) What are their status? Will they most likely end up in stable in the coming future?

Now with Kompact on stable, it would be nice to aim for stable on arcon as well :smile:

Reply:
I mean...I'll continue focusing on nightly Kompact.
Just because Kompact works on stable, doesn't mean I like it that way^^

Reply:
boooo

Reply:
I'm not really sure that the point of the `Fn*` traits is, though...all it gives you a syntax like Scala's `apply` and the ability to pass custom types in places that expect closures without double wrapping (which the compiler should eliminate anyway). They seem very hacky...not sure they'll ever hit stable due to combination of hacky vs. utility...

Reply:
I _thought_ we needed it to use the nice fn-trait-like syntax for our `SafelySendableFn`. Turns out I was mistaken.

Reply:
We only need `unboxed_closures` for that, not `fn_traits`.